keras_cv/layers/preprocessing3d/random_pasting_objects.py

```diff
@@ -0,0 +1,242 @@
+        The first 8 features are [x, y, z, dx, dy, dz, phi, box class].
+
+    Output shape:
+      A tuple of two Tensors (point_clouds, bounding_boxes) with the same shape as input Tensors.
+
+    Arguments:
+      label_index: An optional int scalar sets the target object index.
+        Bounding boxes and corresponding point clouds with box class == label_index will be saved as OBJECT_BOUNDING_BOXES and OBJECT_POINT_CLOUDS.
+        If label index is None, all valid bounding boxes (box class !=0) are used.
+      min_paste_bounding_boxes: A int scalar sets the min number of pasted bounding boxes.
+      max_paste_bounding_boxes: A int scalar sets the max number of pasted bounding boxes.
+      
+    """
+
+    def __init__(
+        self, label_index=None, min_paste_bounding_boxes=0, max_paste_bounding_boxes=10, **kwargs
+    ):
+        super().__init__(**kwargs)
+        if label_index and label_index < 0:
+            raise ValueError("label_index must be >=0.")
+        if min_paste_bounding_boxes < 0:
+            raise ValueError("min_paste_bounding_boxes must be >=0.")
+        if max_paste_bounding_boxes < 0:
+            raise ValueError("max_paste_bounding_boxes must be >=0.")
+        if max_paste_bounding_boxes < min_paste_bounding_boxes:
+            raise ValueError(
+                "max_paste_bounding_boxes must be >= min_paste_bounding_boxes."
+            )
+
+        self._label_index = label_index
+        self._min_paste_bounding_boxes = min_paste_bounding_boxes
+        self._max_paste_bounding_boxes = max_paste_bounding_boxes
+        self._iou_3d = iou_3d.IoU3D()
+
+    def get_random_transformation(
+        self,
+        point_clouds,
+        bounding_boxes,
+        object_point_clouds,
+        object_bounding_boxes,
+        **kwargs
+    ):
+        del point_clouds
+        num_paste_bounding_boxes = self._random_generator.random_uniform(
+            (),
+            minval=self._min_paste_bounding_boxes,
+            maxval=self._max_paste_bounding_boxes,
+        )
+        num_paste_bounding_boxes = tf.cast(num_paste_bounding_boxes, dtype=tf.int32)
+        num_existing_bounding_boxes = tf.shape(bounding_boxes)[1]
+        if self._label_index:
+            object_mask = tf.math.equal(
+                object_bounding_boxes[0, :, 7], self._label_index
+            )
+            object_point_clouds = tf.boolean_mask(
+                object_point_clouds, object_mask, axis=1
+            )
+            object_bounding_boxes = tf.boolean_mask(
+                object_bounding_boxes, object_mask, axis=1
+            )
+        shuffle_index = tf.range(tf.shape(object_point_clouds)[1])
+        shuffle_index = tf.random.shuffle(shuffle_index)
+        object_point_clouds = tf.gather(object_point_clouds, shuffle_index, axis=1)
+        object_bounding_boxes = tf.gather(object_bounding_boxes, shuffle_index, axis=1)
+
+        # Load at most 5 times num_paste_bounding_boxes to check overlaps.
```

Comment:
any reason for at most 5 times?

Reply:
It is an arbitrary number.

Reply:
What's the reasoning here? Maybe add some comment?

Reply:
Add a comment in the docstrings
`We load 5 times max_paste_bounding_boxes to check overlap. `